File: Solutions/han3l/Game Shop/part3/Shop.cpp
#include "Shop.h"

Shop::Shop()
{
}

Shop::~Shop()
{
}

Shop::Shop(std::string name, std::list<Item*> items)
	: name(name), items(items)
{
}

void Shop::ShowItemList()
{
	std::cout << "Welcome to " << this->name << std::endl;
	std::cout << "- Item List -" << std::endl;
	
	for each (auto var in items)
	{
		var->Describe();
	}
}

File: Solutions/LeeInkyu/Game Shop/03/Shop.java
/**
 * Created by Inkyu Lee on 2017-05-30.
 */
public class Shop {
	String name;
	Item[] items;
	int ARR_MAX = 5;


	public Shop(String namePm, Item[] itemsPm)
	{
		this.name = namePm;
		this.items = itemsPm;
	}

	public Shop(String namePm, Item[] itemsPm, int arrMaxPm)
	{
		this.name = namePm;
		this.items = itemsPm;
		this.ARR_MAX = arrMaxPm;
	}

	public void showItemList()
	{
		System.out.println("Welcome to " + this.name + "!");
		for(int i=0; i<ARR_MAX; i++)
		{
			this.items[i].describe();
		}
	}
}
